packages/internals/renderer/src/adapter.js

```diff
@@ -0,0 +1,40 @@
+const {isFunction} = require('lodash');
+const {Validator, File} = require('@frctl/support');
+const {toArray, normalizeExt} = require('@frctl/utils');
+const adapterSchema = require('./adapter.schema');
+
+const _props = new WeakMap();
+
+class Adapter {
+
+  constructor(props) {
+    Validator.assertValid(props, adapterSchema, 'Adapter schema invalid [adapter-invalid]: ');
+    _props.set(this, props);
+  }
+
+  match(file) {
+    if (!File.isFile(file)) {
+      throw new TypeError(`Adapter.match - file argument must be an instance of File [file-invalid]`);
+    }
+    const props = _props.get(this);
+    const match = isFunction(props.match) ? props.match : toArray(props.match).map(ext => normalizeExt(ext));
```

Comment:
Yes that would be better! It was actually that way before I converted it to class but slipped in there somehow. Will move it to the constructor.